sdk/src/epoch_rewards_hasher.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::arithmetic_side_effects)]
```

Comment:
is this for the `u128` downcast to usize, or can we limit it to the tests?

Reply:
fwiw, that usize (target dependent type :vomiting_face:) should probably not be in the api at all. it should be a u64

Reply:
Unless it's highlighting the wrong thing, this is expression that clippy thinks as "can potentially result in unexpected side-effects": `u128::from(u64::MAX).saturating_add(1)`

Reply:
> fwiw, that usize (target dependent type 🤮) should probably not be in the api at all. it should be a u64
I'll not argue that point, but outside the scope of this PR

Reply:
> Known safe built-in types like Wrapping or Saturating, floats, operations in constant environments, allowed types and non-constant operations that won’t overflow are ignored

from [here](https://rust-lang.github.io/rust-clippy/master/index.html#/arithmetic_side_effects). sounds like a linting bug. can we limit the allow to the offending statement instead of the whole module?

Reply:
Yes sure, done